instrumentation/opentelemetry-instrumentation-urllib/src/opentelemetry/instrumentation/urllib/__init__.py

```diff
@@ -248,3 +263,44 @@ def _uninstrument_from(instr_root, restore_as_bound_func=False):
     if restore_as_bound_func:
         original = types.MethodType(original, instr_root)
     setattr(instr_root, instr_func_name, original)
+
+
+def _create_client_histograms(meter) -> Dict[str, Histogram]:
+    histograms = {
+        MetricInstruments.HTTP_CLIENT_DURATION: meter.create_histogram(
+            name=MetricInstruments.HTTP_CLIENT_DURATION,
+            unit="ms",
+            description="measures the duration outbound HTTP requests",
+        ),
+        MetricInstruments.HTTP_CLIENT_REQUEST_SIZE: meter.create_histogram(
+            name=MetricInstruments.HTTP_CLIENT_REQUEST_SIZE,
+            unit="By",
+            description="measures the size of HTTP request messages (compressed)",
+        ),
+        MetricInstruments.HTTP_CLIENT_RESPONSE_SIZE: meter.create_histogram(
+            name=MetricInstruments.HTTP_CLIENT_RESPONSE_SIZE,
+            unit="By",
+            description="measures the size of HTTP response messages (compressed)",
+        ),
+    }
+
+    return histograms
+
+
+def _record_histograms(
+    histograms, metric_attributes, request, response, elapsed_time
+):
+    histograms[MetricInstruments.HTTP_CLIENT_DURATION].record(
+        elapsed_time, attributes=metric_attributes
+    )
+
+    data = getattr(request, "data", None)
+    request_size = 0 if data is None else len(data)
```

Comment:
Will this be a compressed size? Does data always return an iterable?

Reply:
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1399

Reply:
Thanks for the comment, I checked this issue will I wrote the code and the difference is the urllib api works only with bytes, and urllib3 works with bytes and IO(like BytesIO)